versioning_base.py

```diff
@@ -1092,8 +1121,80 @@ def add_branch( pg_conn_info, schema, branch, commit_msg,
+    """
+    pcur = Db(psycopg2.connect(pg_conn_info))
+
+    pcur.execute("SELECT * FROM "+schema+".revisions "
+        "WHERE branch = '"+branch+"'")
+    if not pcur.fetchone():
+        pcur.close()
+        raise RuntimeError("Branch "+branch+" doesn't exist")
+    pcur.execute("SELECT MAX(rev) FROM "+schema+".revisions")
+    [max_rev] = pcur.fetchone()
+    if int(rev_begin) > max_rev or int(rev_begin) <= 0:
+        pcur.close()
+        raise RuntimeError("Revision 1 (begin) "+str(rev_begin)+" doesn't exist")
+    if int(rev_end) > max_rev or int(rev_end) <= 0:
+        pcur.close()
+        raise RuntimeError("Revision 2 (end) "+str(rev_end)+" doesn't exist")
+
+    select_str = ("SELECT "
+    "CASE WHEN "
+        +schema+"."+table+"."+branch+"_rev_begin > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_begin <= "+str(rev_end)+ " "
+        "AND " +schema+"."+table+"."+branch+"_parent IS NULL THEN 'a' "
+    "WHEN (" +schema+"."+table+"."+branch+"_rev_begin > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_end IS NULL "
+        "AND " +schema+"."+table+"."+branch+"_parent IS NOT NULL) "
+        "OR ("+schema+"."+table+"."+branch+"_rev_end >= "+str(rev_end)+" "
+        "AND "+schema+"."+table+"."+branch+"_child IS NOT NULL) "
+        "THEN 'u' "
+    "WHEN " +schema+"."+table+"."+branch+"_rev_end > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_end < "+str(rev_end)+ " "
+        "AND " +schema+"."+table+"."+branch+"_child IS NULL THEN 'd' ELSE 'i' END "
+    "as diff_status, * FROM "+schema+"."+table+ " "
+    "WHERE (" +schema+"."+table+"."+branch+"_rev_begin > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_begin <= "+str(rev_end)+") "
+        "OR (" +schema+"."+table+"."+branch+"_rev_end > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_end <= "+str(rev_end)+ " )")
+
+    pcur.close()
+    return select_str
+
+def rev_view_str(pg_conn_info, schema, table, branch, rev):
+    """REVision_VIEW_STRing
+    Create the SQL view string of the specified revision.
+    Replaces add_revision_view()
+    """
+    pcur = Db(psycopg2.connect(pg_conn_info))
+
+    pcur.execute("SELECT * FROM "+schema+".revisions "
+        "WHERE branch = '"+branch+"'")
+    if not pcur.fetchone():
+        pcur.close()
+        raise RuntimeError("Branch "+branch+" doesn't exist")
+    pcur.execute("SELECT MAX(rev) FROM "+schema+".revisions")
+    [max_rev] = pcur.fetchone()
+    if int(rev) > max_rev or int(rev) <= 0:
+        pcur.close()
+        raise RuntimeError("Revision "+str(rev)+" doesn't exist")
+
+    select_str = "SELECT * FROM "+schema+"."+table
+    #print "select_str = " + select_str
+    where_str = ("("+branch + "_rev_end IS NULL "
+        "OR "+branch+"_rev_end >= "+str(rev) + ") "
+         "AND "+branch+"_rev_begin <= "+str(rev) )
+
+    pcur.close()
+    return [select_str, where_str]
```

Comment:
I don't see the interest of returning the `select_str`, I think it would be clearer to rename the function `where_clause` and just return the `where_str`